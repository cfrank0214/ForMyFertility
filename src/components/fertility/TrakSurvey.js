import React, { Component } from 'react';
import { WebView } from 'react-native';
import { Text, Body, Spinner } from "native-base";

export default class TrakSurvey extends Component {
    constructor(props) {
        super(props);
        this.state = {
        }
    }
    render() {
        return (
            <WebView
                source={{ uri: 'https://trakfertility.tools/api/embed/quiz/3' }}
                useWebKit={true}
                javaScriptEnabled={true}
                renderLoading={this._renderLoading}
                startInLoadingState={true}
            />
        );
    }
    
    _renderLoading = () => {
        return (
            <Body style={{ marginTop: 10 }}>
            <Spinner />
              <Text>
                Loading Trak Survey
            </Text>
            </Body>
        );
    }
}

