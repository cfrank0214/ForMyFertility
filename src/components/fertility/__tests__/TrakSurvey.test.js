/**
 * @jest-environment jsdom
 */
import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TrakSurvey from '../TrakSurvey';
import { WebView } from 'react-native';
import { Text, Body, Spinner } from "native-base";

describe('>>>TrakSurvey --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper
  
  beforeEach(()=>{
      wrapper = shallow(<TrakSurvey />)
      
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <TrakSurvey />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('+++ render the DUMB component', () => {
     expect(wrapper.length).toEqual(1)
  }); 
 it('contains 1 <WebView /> component', () => {
  expect(wrapper.find(WebView).length).toBe(1);
}); 
});