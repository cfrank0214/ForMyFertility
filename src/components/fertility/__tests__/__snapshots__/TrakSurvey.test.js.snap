// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>TrakSurvey --- Shallow Render REACT COMPONENTS renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWKWebView
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    source={
      Object {
        "uri": "https://trakfertility.tools/api/embed/quiz/3",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
        Object {
          "flex": 0,
          "height": 0,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        },
        Object {
          "marginTop": 10,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#45D56E"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "height": 80,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      Loading Trak Survey
    </Text>
  </View>
</View>
`;
